pkg/controllers/core/deployer_controller.go

```diff
@@ -103,11 +101,39 @@ func (r *DeployerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return result, err
 }
 
+func (r *DeployerReconciler) getAndTrackConfigMap(ctx context.Context, log logr.Logger, deployer *corev1alpha1.Deployer) (*corev1.ConfigMap, error) {
```

Comment:
No need to couple this method to the deployer. Any object can be tracked.

```suggestion
func (r *DeployerReconciler) getAndTrackConfigMap(ctx context.Context, log logr.Logger, obj runtime.Object) (*corev1.ConfigMap, error) {
```

It may also be worth pulling this method out of this reconciler, because it could be generally useful.

Reply:
Perhaps as a separate change? Right now, we're tracking a specific configmap `riff-core-settings`.  As the refactor should probably pull out the `getAndTrackConfigMap` function from the other controllers as well, I want to hold out on that and keep this change limited to just the core deployer.

Reply:
I'd still like to decouple this method from the Deployer type since the runtime.Object interface has everything we need.

Perhaps the method should also be renamed to `getAndTrackSettings` since the method doesn't get any arbitrary config map.